video_xblock/tests/mocks.py

```diff
@@ -0,0 +1,526 @@
+    """
+    Wistia default transcripts mock class.
+    """
+    _expected = [
+        {
+            'lang': u'en',
+            'url': 'url_can_not_be_generated',
+            'text': u'http://video.google.com/timedtext?lang=en&name=&v=set_video_id_here',
+            'label': u'English'
+        },
+        {
+            'lang': u'uk',
+            'url': 'url_can_not_be_generated',
+            'text': u'http://video.google.com/timedtext?lang=uk&name=&v=set_video_id_here',
+            'label': u'Ukrainian'
+        }
+    ]
+
+    _default_transcripts = [
+        {'english_name': u'English', 'language': u'eng',
+         'text': 'http://video.google.com/timedtext?lang=en&name=&v=set_video_id_here'},
+        {'english_name': u'Ukrainian', 'language': u'ukr',
+         'text': 'http://video.google.com/timedtext?lang=uk&name=&v=set_video_id_here'}
+    ]
+
+    event_types = [
+        'request_data_exception', 'success_and_data_lang_code',
+        'success_and_data_lang_code_exception', 'success_no_data', 'returned_not_found']
+
+    event_results = {
+        'request_data_exception': {
+            'default_transcripts': [],
+            'message': 'No timed transcript may be fetched from a video platform.'
+        },
+        'success_and_data_lang_code': {
+            'default_transcripts': _expected,
+            'message': ''
+        },
+        'success_and_data_lang_code_exception': {
+            'default_transcripts': _default_transcripts,
+            'message': 'LanguageReverseError'
+        },
+        'success_no_data': {
+            'default_transcripts': [],
+            'message': 'For now, video platform doesn\'t have any timed transcript for this video.'
+        },
+        'returned_not_found': {
+            'default_transcripts': [],
+            'message': 'doesn\'t exist.'
+        }
+    }
+
+    to_return = ['default_transcripts', 'message']
+
+    def get(self, event):
+        """
+        Substitute requests.get method.
+        """
+        if event == 'request_data_exception':
+            self.side_effect = self.mock()
+            return self
+        elif event == 'success_and_data_lang_code':
+            self.return_value = Response(status_code=200, body=json.dumps(self._default_transcripts))
+        elif event == 'success_and_data_lang_code_exception':
+            default_transcripts = copy(self._default_transcripts)
+            default_transcripts[0]['language'] = 'en'
+            self.return_value = Response(status_code=200, body=json.dumps(default_transcripts))
+        elif event == 'success_no_data':
+            self.return_value = Response(status_code=200, body='[]')
+        elif event == 'returned_not_found':
+            self.return_value = Response(status_code=404, body='{}')
+        return lambda x: self.return_value
+
+
+class YoutubeDownloadTranscriptMock(BaseMock):
+    """
+    Youtube download default transcript mock class.
+    """
+    _xml = (
```

Comment:
Structure of xml should be different, here is a sample:

`'<?xml version="1.0" encoding="utf-8" ?><transcript><text start="0" dur="1.68">[INTRODUZIONE]</text><text start="3.093" dur="3.3">Oggi voglio parlare di\nsottotitoli, di nuovo.</text><text start="6.4" dur="4.412">Come sapete, io sono una grande\nsostenitrice dei sottotitoli su YouTube.</text><text start="10.812" dur="2.479">Forse me la canto e me la suono da sola un po&amp;#39;,</text><text start="13.291" dur="5.56">ma alcuni dei pi\xf9 grandi YouTuber mi\nhanno citato, e hanno sottotitolato i loro video</text></transcript>'`